app/Helpers/database/leaderboard.php

```diff
@@ -154,197 +134,114 @@ function removeLeaderboardEntry(User $user, int $lbID, ?string &$score): bool
-                          LEFT JOIN LeaderboardDef AS lbd ON lbd.ID = le.leaderboard_id
-                          WHERE ua.Untracked = 0 AND le.leaderboard_id = $lbID) InnerTable
-                          WHERE InnerTable.User = '$user'";
-
-                $dbResult = s_mysql_query($query);
-                if ($dbResult !== false) {
-                    if (mysqli_num_rows($dbResult) > 0) {
-                        // should be 1 or 0?.. I hope...
-                        $db_entry = mysqli_fetch_assoc($dbResult);
-                        $db_entry['DateSubmitted'] = strtotime($db_entry['DateSubmitted']);
-                        $db_entry['Score'] = (int) $db_entry['Score'];
-                        $db_entry['Rank'] = (int) $db_entry['UserRank'];
-                        // @phpstan-ignore-next-line
-                        unset($db_entry['UserRank']);
-                        // @phpstan-ignore-next-line
-                        $db_entry['Index'] = (int) $db_entry['UserIndex'];
-                        // @phpstan-ignore-next-line
-                        unset($db_entry['UserIndex']);
-                        $entries[] = $db_entry;
-                    }
-                } else {
-                    log_sql_fail();
-                }
-            }
+        $index++;
+    }
 
-            $retVal['Entries'] = $entries;
-        } else {
-            log_sql_fail();
+    // Currently only used for appending player to the end on website leaderboard pages
+    if ($userFound === false && $user && !$nearby) {
+        $entry = getLeaderboardUserEntry($leaderboard, $user);
+        if ($entry) {
+            $retVal['Entries'][] = $entry;
         }
     }
 
     return $retVal;
 }
 
-function getLeaderboardUserPosition(int $lbID, string $user, ?int &$lbPosition): bool
+function getLeaderboardUserEntry(Leaderboard $leaderboard, string $user): ?array
 {
-    sanitize_sql_inputs($user);
-
-    $query = "SELECT UserIndex FROM
-                         (SELECT ua.User, le.score, le.updated_at,
-                          CASE WHEN lbd.LowerIsBetter = 0
-                          THEN ROW_NUMBER() OVER(ORDER BY le.score DESC, le.updated_at ASC)
-                          ELSE ROW_NUMBER() OVER(ORDER BY le.score ASC, le.updated_at ASC) END AS UserIndex
-                          FROM leaderboard_entries AS le
-                          LEFT JOIN UserAccounts AS ua ON ua.ID = le.user_id
-                          LEFT JOIN LeaderboardDef AS lbd ON lbd.ID = le.leaderboard_id
-                          WHERE ua.Untracked = 0 AND le.leaderboard_id = $lbID) InnerTable
-                          WHERE InnerTable.User = '$user'";
-
-    $dbResult = s_mysql_query($query);
-    if ($dbResult !== false) {
-        $db_entry = mysqli_fetch_assoc($dbResult);
+    $userEntry = $leaderboard->entries(includeUnrankedUsers: true)
+        ->whereHas('user', function ($query) use ($user) {
+            $query->where('User', '=', $user);
+        })
```

Comment:
If `$user` is a `User` instead of a `string`, we should be able to query from `user_id` here.

Reply:
done.